### DISCUSSION TITLE
Renaming many collections at once doesn't work

### DISCUSSION THREAD

**Discussion by soopax** [ppy/osu#23739] [Q&A] [answered]
# Steps to reproduce
1. Open the "Manage collections" window
2. Create two new collections "test 1" and "test 2"
3. Click "test 1" and rename it to "test 1 foo", but don't press `Enter`
4. Click "test 2" and rename it to "test 2 bar" and *now* press `Enter`

## Expected result
* test 1 foo
* test 2 bar

## Actual result
* test 1
* test 2 bar

**Comment by monochrome22** [comment]
Expecting the previous edit to save while pressing enter on the current edit is weird UX overall.

It might be a good idea to have the saving for all changes happen when the Manage Collections window is closed. Pressing Enter would only finish text input and pressing Esc would revert changes to the current text box and deselect it. 

Saving when exiting would also take care of this problem:

https://github.com/ppy/osu/assets/122969285/35349c66-c144-4ea0-b323-cbee82996189

(Different texts are displayed in Manage Collections and the collection list if you leave without pressing Enter.)

**Comment by peppy** [comment, marked as answer]
https://github.com/ppy/osu/pull/23741
